macos/resource_compression/dcmp_0.ksy

```diff
@@ -0,0 +1,525 @@
+meta:
+  id: dcmp_0
+  title: Compressed Macintosh resource data, Apple `'dcmp' (0)` format
+  application: Mac OS
+  license: MIT
+  ks-version: "0.8"
+  imports:
+    - dcmp_variable_length_integer
+  endian: be
+doc: |
+  Compressed resource data in `'dcmp' (0)` format,
+  as stored in compressed resources with header type `8` and decompressor ID `0`.
+
+  The `'dcmp' (0)` decompressor resource is included in the System file of System 7.0 and later.
+  This compression format is used for most compressed resources in System 7.0's files.
+  This decompressor is also included with and used by some other Apple applications,
+  such as ResEdit.
+
+  This compression format supports some basic general-purpose compression schemes,
+  including backreferences to previous data,
+  run-length encoding,
+  and delta encoding.
+  It also includes some types of compression tailored specifically to Mac OS resources,
+  including a set of single-byte codes that correspond to entries in a hard-coded lookup table,
+  and a specialized kind of delta encoding for segment loader jump tables.
+
+  Almost all parts of this compression format operate on units of 2 or 4 bytes.
+  As a result,
+  it is nearly impossible to store data with an odd length in this format.
+  To work around this limitation,
+  odd-length resources are padded with an extra byte before compressing them with this format.
+  This extra byte is ignored after decompression,
+  as the real (odd) length of the resource is stored in the compressed resource header.
+
+  The `'dcmp' (1)` compression format (see dcmp_1.ksy) is very similar to this format,
+  with the main difference that it operates mostly on single bytes rather than two-byte units.
+doc-ref: 'https://github.com/dgelessus/python-rsrcfork/tree/master/rsrcfork/compress/dcmp0.py'
+seq:
+  - id: chunks
+    type: chunk
+    repeat: until
+    repeat-until: _.tag == 0xff
+    doc: |
+      The sequence of chunks that make up the compressed data.
+types:
+  chunk:
+    seq:
+      - id: tag
+        type: u1
+        doc: |
+          The chunk's tag byte.
+          This controls the structure of the body and the meaning of the chunk.
+      - id: body
+        type:
+          switch-on: |
+            tag >= 0x00 and tag <= 0x1f ? "literal"
```

Comment:
I wonder if it is more efficient to encode as numbers. I mean enum members.

Reply:
An enum works quite well here, thanks for the suggestion!

It could be implemented even more nicely with kaitai-io/kaitai_struct#489, but we don't have any feature like that yet.